### PR TITLE
MAIN - Use pika pack to build the npm package

### PR DESCRIPTION
**Pull request by Weetbix**
👋  This PR switches over to using pika pack to build the npm package. 

https://www.pikapkg.com/blog/introducing-pika-pack/

Currently we are bundling the package with webpack and serving it as a single minified dist file. This is a bit odd for a library, as the consumer cannot chose the transpliation level, and there is a _lot_ of work associated with correctly bundling for the many different types of targets we should be supporting. 

This PR: 
- Adds pika as a dependency and uses it to build our package into the `./pkg` folder, which is then the published folder.
- Changes the library API so we now support named imports from the library, which enables tree shaking in consumer applications. ie `import { create } from "@vivy/krypt-web"`
- Moves from karma/webpack/mocha/chai/sinon test stack to `karmatic` which uses jasmine. As you know we need to run our tests in the browser, but the test runners that support this require bundling/transpiling of the source code, but our package is no longer bundled into 1 file. We do not want to spend time maintaining a webpack bundle just for running the tests, so for that reason I've moved to karmatic (https://github.com/developit/karmatic). Which is basically a configless headless browser stack (using webpack/karma/puppeteer), this allows us to get rid of *all* webpack/babel configs. It also moves us closer to a jest syntax (as jest is based on jasmine). It would have been ideal to use jest here but I couldn't find an package or environment that would allow us to run our actual tests inside the browser environment. 
- Moves to use our company eslint config 

I have tested this locally with an application that uses the library by following these steps:
- build the package `npm run build:package`
- package it into a local file `cd ./pkg; npm pack` 
- installed that into an app that uses it `npm i <path to file output>`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,51 +7,55 @@
...
-    "karma-webpack": "^3.0.5",
-    "mocha": "^5.2.0",
+    "karmatic": "^1.3.1",
     "prettier": "^1.16.1",
     "pretty-quick": "^1.10.0",
     "raw-loader": "1.0.0",
     "rimraf": "^2.6.3",
-    "webpack": "^4.29.0",
-    "webpack-cli": "^3.2.3"
+    "webpack": "^4.32.0"
```
Comment (paschalidi) [inline package.json]: 🤔 wondering, why we still need webpack?

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import create from "./lib/factory";
+
+export { create };
```
Comment (paschalidi) [inline src/index.js]: cant we make something like `export {default as create} from "./lib/factory";` here?

File: package.json (hunk 1 above)
Comment (Weetbix) [reply package.json]: Its a peer dependency of karmatic so is expected to be installed by the consumer, check out the installation instructions here: https://github.com/developit/karmatic

File: src/index.js (hunk 2 above)
Comment (Weetbix) [reply src/index.js]: Nah we can't, there is a proposal but its early stages https://github.com/tc39/proposal-export-default-from so better to leave it :)

**Pull request merged**